lib/experimental/block-hooks.php

```diff
@@ -325,17 +330,17 @@ function gutenberg_serialize_blocks( $blocks ) {
 }
 
 /**
- * Register the `auto_insert` field for the block-types REST API controller.
+ * Register the `block_hooks` field for the block-types REST API controller.
  *
  * @return void
  */
-function gutenberg_register_auto_insert_rest_field() {
+function gutenberg_register_block_hooks_rest_field() {
 	register_rest_field(
 		'block-type',
-		'auto_insert',
+		'block_hooks',
 		array(
 			'schema' => array(
-				'description'       => __( 'Block types that may be automatically inserted near this block and the associated relative position where they are inserted.', 'gutenberg' ),
+				'description'       => __( 'This block is automatically inserted near any occurence of the block types used as keys of this map, into a relative position given by the corresponding value.', 'gutenberg' ),
```

Comment:
See https://github.com/WordPress/gutenberg/pull/52969#discussion_r1315074462:

> BTW while working on #54147, I discovered that my original description was inaccurate [...]: The field isn't for block types that are automatically inserted next to this block, but the other way around: we're inserting this block next to those block types 😬 

I'll update, probably as part of #54147.

cc/ @dmsnell for wording 😊